src/actions.cpp

```diff
@@ -214,7 +214,17 @@ ReturnValue Actions::internalUseItem(const std::shared_ptr<Player>& player, cons
 		// depot container
 		if (const auto& depot = container->getDepotLocker()) {
 			container = player->getDepotLocker();
-			container->setParent(depot->getParent()->getTile());
+
+			std::shared_ptr<Tile> parentTile = nullptr;
+			if (const auto& item = depot->getParent()->asItem()) {
+				parentTile = item->getTile();
+			} else if (const auto& creature = depot->getParent()->asCreature()) {
+				parentTile = creature->getTile();
+			} else if (const auto& tile = depot->getParent()->asTile()) {
+				parentTile = tile;
+			}
```

Comment:
This seems to repeat itself quite often. We can still have `getTile` doing the same thing that this snippet does, it's already miles better to have `asTile()` specifically to cast `Thing` to `Tile`

Reply:
I agree, initially thought about having an internal `getTile` in Thing, but that would cause problems with the inheriting classes, so ill keep as virtual and maintain the previous impl